File: KLASA_3/ROZDZIAL_1/LEKCJA_1/PYTHON/ZAD1.py
from stos import Stos

def sprawdz_wyrazenie(wyrazenie_nawiasowe):
    stos = Stos()
    for nawias in wyrazenie_nawiasowe:
        if nawias == '(':
            stos.push(nawias)
        elif nawias == ')':
            if stos.isEmpty():
                return False
            else:
                if stos.peek() != '(':
                    return False
                else:
                    stos.pop()
    return stos.isEmpty()


print('TAK' if sprawdz_wyrazenie(input('Podaj wyrazenie nawiasowe: ')) else 'NIE')

File: KLASA_3/ROZDZIAL_1/LEKCJA_1/C++/ZAD1.cpp
#include <iostream>
#include <stack>

using namespace std;

bool czy_poprawny(string wyrazenie)
{
    stack<char> stos;

    for(char nawias: wyrazenie){
        if(nawias == '('){
            stos.push(nawias);
        }else if(nawias == ')'){
            if(stos.empty())return false;
            if(stos.top() != '(')return false;
            stos.pop();
        }
    }

    return stos.size() == 0;
}

int main ()
{
    string wyrazenie;
    cout << "Podaj wyrazenie nawiasowe: ";
    cin >> wyrazenie;

    cout << (czy_poprawny(wyrazenie) ? "TAK" : "NIE");
    return 0;
}
